synapse/cortex.py

```diff
@@ -4587,7 +4587,8 @@ async def addCronJob(self, cdef):
         except KeyError:
             raise s_exc.BadConfValu('Unrecognized time unit')
 
-        cdef['iden'] = s_common.guid()
+        if not cdef.get('iden'):
+            cdef['iden'] = s_common.guid()
 
         return await self._push('cron:add', cdef)
 
```

Comment:
Lets discuss what to do with a collision.  The previous code assumes any collisions are a nexs replay artifact and we might need to do a bit more.